pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java

```diff
@@ -204,8 +204,8 @@ public String segmentCommitEnd(@QueryParam(SegmentCompletionProtocol.PARAM_INSTA
       return SegmentCompletionProtocol.RESP_FAILED.toJsonString();
     }
 
-    SegmentMetadataImpl segmentMetadata = extractMetadataFromLocalSegmentFile(segmentName,
-            ControllerConf.getUriFromPath(segmentLocation).getPath());
+    SegmentMetadataImpl segmentMetadata = extractMetadataFromSegmentFile(segmentName,
+            ControllerConf.getUriFromPath(segmentLocation));
```

Comment:
segmentLocation should already be a well-formed URI.  See the segmentUpload endpoint.

Reply:
I still prefer to use a URI to pass into 
extractMetadataFromSegmentFile(.,.) because PinotFS expects an URI in general.  Construct a URI directly instead of calling 
ControllerConf.getUriFromPath (which is essentially a thin wrapper).

Reply:
This is a URI returned by the controller in the segmentUpload call. The servers cannot send some random URIs to commit. So, to PinotFS, you should be able to pass the URI from new URI(segmentLocation)

Reply:
@mcvsubbu My [latest commit ](https://github.com/apache/incubator-pinot/pull/3877/commits/abba120a37dd8c06fb76bcf885a7e7ee45b7f27b) already passed in 
new URI(segmentLocation) as suggested. Please take a look.